projects/103_homework/src/main.c

```diff
@@ -0,0 +1,60 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// interrupt call back function
+void interrupt_callback(const GpioAddress *address, void *context) {
+  GpioAddress *ADC_address = context;
+
+  uint16_t ADC_reading = 0;
+  if (adc_read_raw_pin(*ADC_address, &ADC_reading) == STATUS_CODE_OK) {
+    LOG_DEBUG("Output: &d\n", ADC_reading);
+  } else {
+    LOG_DEBUG("Failed to read.");
+  }
+}
+
+int main(void) {
+  gpio_it_init();
```

Comment:
If you have a look at each library you will see that there is a hierarchy of calling init functions.
For example, in gpio_it.h: 
`// Requires GPIO and interrupts to be initialized.`
So you must call gpio_init() and interrupt_init() before gpio_it_init()